includes/abstracts/abstract-wc-data.php

```diff
@@ -148,6 +151,26 @@ public function get_data() {
 	}
 
 	/**
+	 * Returns array of expected data keys for this object.
+	 *
+	 * @since 2.7.0
+	 * @return array
+	 */
+	public function get_data_keys() {
+		return array_keys( $this->data );
+	}
+
+	/**
+	 * Returns all "extra" data keys for an object (for sub objects like product types).
+	 *
+	 * @since 2.7.0
+	 * @return array
+	 */
+	public function get_extra_data_keys() {
+		return array_keys( $this->extra_data );
```

Comment:
Yeah I think we can make that standard for all classes. And it's required to isolate just the extra fields easily (button_text or product_url for external products) -- which I use in the read().